src/coreclr/tools/superpmi/superpmi-shim-collector/CMakeLists.txt

```diff
@@ -5,7 +5,6 @@ remove_definitions(-D_UNICODE)
 
 add_definitions(-DFEATURE_NO_HOST)
 add_definitions(-DSELF_NO_HOST)
-add_definitions(-DFEATURE_ENABLE_HARDWARE_EXCEPTIONS)
```

Comment:
This was added in #69494. We want this in the shim to properly be able to collect SPMI collections on JIT crashes and diagnose them.

Reply:
@jkotas  I guess this feature isn't as unused as I thought

Reply:
> This was added in https://github.com/dotnet/runtime/pull/69494. We want this in the shim to properly be able to collect SPMI collections on JIT crashes and diagnose them.

Does it actually work?

We tried to do the same in DAC/SOS. We found that it does not work reliably. The C/C++ codegen is not expecting async exceptions on non-Windows.

If we want to keep for superpmi, I would move it under superpmi. It is not a feature that we can use in a shipping code.

Reply:
Yes, it works/worked fine. We don't need much here, we just store a file and allow the exception to propagate afterwards, so maybe that's why it worked for us.

> If we want to keep it for superpmi, I would move it under superpmi. It is not a feature that we can use in any shipping code.

Sounds fine to me. We might get by with just a simple signal handler that saves the collection on SIGSEGV but otherwise leaves the hardware exception unhandled.

Reply:
Ok, I might try something out with a separate thread that gets woken up by a signal handler when the main thread SEGVs.

@jakobbotsch Just to make sure I understand how things work today with the hardware exception handler.  It's this chunk of code that does it?   You compile with your own PAL that is built with the feature flag.  The rest of the runtime/JIT is consumed as prebuilt shared libraries.  So the CatchHardwareExceptionHolder is only active on this one thread just for the duration of this single try/catch/finally block?

https://github.com/dotnet/runtime/blob/53f461449ff996884087910b97ec4509a558b070/src/coreclr/tools/superpmi/superpmi-shim-collector/icorjitcompiler.cpp#L126-L144

Reply:
Yes, that's the main bit with the finally we want to run even on segfault. The shim is a shared library that sits between the normal coreclr and a specified JIT and intercepts and records the traffic between them.

Normal set up looks something like this:
```
DOTNET_JitName=libsuperpmi-shim-collector.so
SuperPMIShimLogPath=/path/to/folder/to/record/into
SuperPMIShimPath=/path/to/real/libclrjit.so
./corerun/dotnet some_app.dll
```
This should create .mc files in the directory specified. The logic here is trying to ensure those .mc files get created with the traffic seen between the JIT and VM up until the seg fault happened.

The .mc files are replayable via the superpmi executable in Core_Root with something like `./superpmi path/to/libclrjit.so path/to/collection.mc`